src/controllers/challenge.controllers.js

```diff
@@ -15,4 +15,71 @@ export const createChallenge = async (req, res) => {
   }
 };
 
-export default { createChallenge };
+// Get all challenges
+export const getChallenges = async (req, res) => {
+  try {
+    const data = await Challenge.find();
+    // TODO: Filter out challenges that the creater has left
+    res.status(200).json({ data, error: null });
+  } catch (error) {
+    res.status(500).json({ data: null, error: JSON.stringify(error) });
+  }
+};
+
+// Get challenge by id
+export const getChallengeById = async (req, res) => {
+  try {
+    const { id } = req.params;
+    const data = await Challenge.findById(id);
+    res.status(200).json({ data, error: null });
+  } catch (error) {
+    res.status(500).json({ data: null, error: JSON.stringify(error) });
+  }
+};
+
+// Add a member to a challenge
+export const addMember = async (req, res) => {
+  try {
+    const { id } = req.params;
+    const { user_id, joined_at, points, left_at } = req.body;
+    const challenge = await Challenge.findById(id);
+    if (!challenge) {
+      return res.status(404).json({ data: null, error: "Challenge not found" });
+    }
+    challenge.members.push({ user_id, joined_at, points, left_at });
```

Comment:
the left_at field will be controlled by us internally, so you can set it by default as null, and remove the value passed here.

Reply:
resolved: https://github.com/LangaraBeetles/weaup-api/pull/8/commits/9f44b09f4296103306467228995c09ed0edd8d8d